File: Stack/Stack.java

public class Stack {

	int stackSize;
	int arr[];
	int top;

	Stack(int stackSize) {
		this.stackSize = stackSize;
		this.arr = new int[stackSize];
		this.top = -1;
	}

	public void push(int ele) {
		if (!isFull()) {
			top++;
			arr[top] = ele;
			System.out.println("Pushing :" + ele);
		} else {
			System.out.println("Stack is full.");
		}
	}

	public int pop() {
		if (!isEmpty()) {
			int returnedTop = top;
			top--;
			System.out.println("Popping :" + arr[returnedTop]);
			return arr[returnedTop];

		} else {
			System.out.println("Stack is empty !");
			return -1;
		}
	}

	public int peek() {
		if (!this.isEmpty())
			return arr[top];
		else {
			System.out.println("Stack is Empty");
			return -1;
		}
	}

	public boolean isEmpty() {
		return (top == -1);
	}

	public boolean isFull() {
		return (stackSize - 1 == top);
	}

	public static void main(String[] args) {
		Stack StackObj = new Stack(10);
		StackObj.pop();
		System.out.println("Started pushing...");
		StackObj.push(1);
		StackObj.push(20);
		StackObj.push(9);
		StackObj.push(14);
		System.out.println("Started poping...");
		StackObj.pop();
		StackObj.pop();
		StackObj.pop();
	}
}

File: c-programs-main/stack.c
#include <stdio.h>

#define LENGTH 10

int top = -1;
int stack[LENGTH];

void push_stack(int item){
	if(top==LENGTH){
		printf("Stack is full\n\a");
	}
	else{
		top+=1;
		stack[top] = item;
	}
}

int pop(){
	if(top==-1){
		printf("Stack is Empty\n\a");
	}
	else{
		top-=1;
		return stack[top+1];
	}
}

void isEmpty(){
	if(top==-1)
		printf("Yes!!\n\a");
	else
		printf("No!!\n\a");
}

void isFull(){
	if(top==LENGTH)
		printf("Yes!!\n\a");
	else
		printf("No!!\n\a");
}

int displayStack(){
	if(top==-1){
		printf("Empty Stack!!\n\a");
		return(0);
	}
	printf("%d  <----head\n",stack[top]);
	for(int i = top-1;i>-1;--i){
		printf("%d\n",stack[i]);
	}
}


int main(){
	push_stack(12);
	push_stack(23);
	push_stack(67);
	push_stack(1128);
	push_stack(444);
	pop();
	displayStack();
	return(0);
}